discovery-infra/tests/base_test.py

```diff
@@ -34,33 +36,34 @@ def get_nodes(self) -> Callable:
         nodes_data = dict()
 
         @JunitTestCase()
-        def get_nodes_func(config: TerraformConfig = TerraformConfig()):
+        def get_nodes_func(config: Optional[TerraformConfig] = None):
+            if not config:
+                config = TerraformConfig()
+
             if "nodes" in nodes_data:
                 return nodes_data["nodes"]
 
-            net_asset = NetworkAssets()
-            config.net_asset = net_asset.get()
+            with file_lock_context(consts.NODES_DEPLOYMENT_LOCK_FILE):
```

Comment:
Why not inside the Assets object?
Every time someone will use it he needs to remember to lock the file

Reply:
I must have a lock here. If two NetAssets are created simultaneously they can try to allocate the same resources.